### PR TITLE
[BEAM-4559] Split post-commit jobs into automatically triggered and ghprb phrase triggered

### PR DESCRIPTION
**Pull request by Ardagan**
This change separates our post-commit jenkins jobs into two separate jobs:
One job is for test runs that are executed on master HEAD and are triggered automatically based on time or on new commit
Other job is for test runs that are triggered manually on PR change.

It covers one of work items on following design doc:
https://docs.google.com/document/d/1sczGwnCvdHiboVajGVdnZL0rfnr7ViXXAebBAf_uQME

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

### CODE REVIEW

File: .test-infra/jenkins/JobBuilder.groovy [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  private def jobDefinition;
+  private def job;
+
+  private JobBuilder(scope, jobDefinition = {}) {
+    this.scope = scope
+    this.jobDefinition = jobDefinition
+    this.job = null
+  }
+
+  static void postCommitJob(nameBase,
```
Comment (kennknowles) [inline .test-infra/jenkins/JobBuilder.groovy]: With this number of arguments, consider named arguments: http://docs.groovy-lang.org/latest/html/documentation/#_named_arguments

(it is a bit funky)

**Comment by Ardagan** [comment]
@swegner

**Comment by Ardagan** [comment]
Tested best I could via Jenkins on Docker. Jobs look valid.

**Comment by kennknowles** [comment]
I'll merge and any improvements can be a follow-up.

**Comment by kennknowles** [comment]
By the way, can you share any notes about testing Jenkins via Docker? I have never done it, and I bet they would be generally useful.

**Comment by Ardagan** [comment]
Run Seed Job

**Pull request merged**